modules/GanneffServ.rb

```diff
@@ -221,6 +230,50 @@ def DEL(client, parv = [])
     true
   end # def DEL
 
+# ------------------------------------------------------------------------
+
+  # Protect users from collatoral damage
+  def PROTECT(client, parv = [])
+    parv[1].downcase!
+    debug(LOG_DEBUG, "#{client.name} called PROTECT and the params are #{parv.join(",")}")
+
+    requested_pattern = parv[1]
+    pattern = irc_pattern_to_regex(requested_pattern)
+    reason = parv[2]
+
+    ret = DB.execute_nonquery(@dbq['INSERT_PROTECT'], 'iiss', client.nick.account_id,
```

Comment:
Put the original pattern in the DB, rather than the regex. This is important for the listing later, as looking at IRC masks is nicer than regexes, unless we've specifically given a regex.